### ISSUE TITLE
Consider using a custom package type for addins / modules

### ISSUE THREAD

**Issue by pascalberger** [cake-contrib/Home#66]
We should consider using a [custom package type](https://docs.microsoft.com/en-us/nuget/create-packages/set-package-type) for Cake addins, modules and recipes. 

This could help in discovery of these packages from nuget.org.

**Comment by augustoproiete** [comment]
### TL;DR;

Custom package types work well with Cake build scripts, but with Frosting it would impose limitations in the way a package can be installed. Namely, it's not possible to install packages with custom types via the Visual Studio User Interface... (it only works if you edit the .csproj manually yourself and type/paste a `<PackageReference ... />`.

For now we've decided to [propose a feature request to the NuGet team](https://github.com/NuGet/Home/issues/10468) which would allow the installation of NuGet packages with custom types on Frosting projects without limitations, and revisit this issue in the future.

---

### Long version

As as proof-of-concept  I created three NuGet packages and published to nuget.org:

| | Package ID | Description | Package Type |
|:---:| --- | --- | --- |
| :package: | [Cake.PackageType.Addin](https://www.nuget.org/packages/Cake.PackageType.Addin/) | Sample Cake addin with `Hello` aliases | `CakeAddin` |
| :package: | [Cake.PackageType.Module](https://www.nuget.org/packages/Cake.PackageType.Module/) | Sample Cake module with a `hello` scheme for the `#tool` directive |  `CakeModule` |
| :package: | [Cake.PackageType.Recipe](https://www.nuget.org/packages/Cake.PackageType.Recipe/) | Sample Cake recipe with a couple of helper methods | `CakeRecipe` |

The source code of the packages above is available on [here](https://github.com/augustoproiete-sandbox/cake-extension-custom-package-types).

---

### Test Results

| | Test | Result |
|:---:| --- | --- |
| :white_check_mark: | List packages via NuGet Gallery | Works as expected / No difference in behavior |
| :white_check_mark: | List packages via VS NuGet Package Manager UI | Works as expected / No difference in behavior |
| :white_check_mark: | List packages via NuGet API | Works as expected / No difference in behavior |
| :white_check_mark: | List packages via NuGet.exe | Works as expected / No difference in behavior |
| :white_check_mark: | Cake script: Use addin in a Cake build script | Works as expected / No difference in behavior |
| :white_check_mark: | Cake script: Use module in a Cake build script | Works as expected / No difference in behavior |
| :white_check_mark: | Cake script: Use recipe in a Cake build script | Works as expected / No difference in behavior |
| :white_check_mark: | Frosting: Add package to `.csproj` via `PackageReference` | Works as expected / No difference in behavior |
| :white_check_mark: | Frosting: Add package to `.csproj` via `dotnet add package` | Works as expected / No difference in behavior |
| :x: | Frosting: Add package to `.csproj` via VS Package Manager UI | Error / package type not supported |
| :x: | Frosting: Add package to `.csproj` via `Install-Package` from VS | Error / package type not supported |

---

#### Screenshots

<details open>
<summary>Visual Studio NuGet Package Manager UI</summary>
<p>

![image](https://user-images.githubusercontent.com/177608/105110824-e1366f00-5a95-11eb-97ed-4ed6193d2991.png)

</p>
</details>

<details open>
<summary>Error messages in the failed tests</summary>
<p>

![image](https://user-images.githubusercontent.com/177608/105111729-f01e2100-5a97-11eb-9ec0-e634a5b9c375.png)

![image](https://user-images.githubusercontent.com/177608/105111734-f2807b00-5a97-11eb-8328-53ea62f00929.png)

![image](https://user-images.githubusercontent.com/177608/105111741-f6ac9880-5a97-11eb-9719-c5ceb6d4dc90.png)

</p>
</details>

### Conclusion

For now we've decided to [propose a feature request to the NuGet team](https://github.com/NuGet/Home/issues/10468) which would allow the installation of NuGet packages with custom types on Frosting projects without limitations, and revisit this issue in the future.

N.B.: Given that the limitation is only with Frosting, if we wanted to start using custom package types for Recipes today, we could.
